docs/core/diagnostics/metrics-instrumentation.md

```diff
@@ -103,33 +103,89 @@ to install it:
-    hats-sold (Count / 1 sec)                          4
+[HatCo.Store]
+    hatco.store.hats_sold (Count / 1 sec)                          4
 ```
 
 As expected, you can see that HatCo store is steadily selling 4 hats each second.
 
+## Getting a Meter via Dependency Injection
+
+In the example above we obtained the Meter by constructing it with `new` and assigning it to a static field. Using statics this way is not a good approach when using Dependency
+Injection (DI). In code that uses DI such as ASP.NET Core, or apps with [Generic Host](../extensions/generic-host.md) create the Meter object using
+<xref:System.Diagnostics.Metrics.IMeterFactory>. Starting in .NET 8 hosts will automatically register <xref:System.Diagnostics.Metrics.IMeterFactory> in the service container
+or you can manually register the type in any <xref:Microsoft.Extensions.DependencyInjection.IServiceCollection> by calling <xref:Microsoft.Extensions.DependencyInjection.MetricsServiceExtensions.AddMetrics%2A>.
+The meter factory integrates metrics with DI, keeping Meters in different service collections isolated from each other even if they are using an identical name. This is
+especially useful for testing so that multiple tests running in parallel will only observe measurements produced from within the same test case.
+
+To obtain a Meter in a type designed for DI, add an <xref:System.Diagnostics.Metrics.IMeterFactory> parameter to the constructor, then call
+<xref:System.Diagnostics.Metrics.MeterFactoryExtensions.Create%2A>. This example shows adding IMeterFactory to an ASP.NET Core MVC controller type:
+
+```cs
+public class HatStoreController : Controller
+{
+    private readonly Counter<int> _hatsSold;
+
+    public HatStoreController(IMeterFactory meterFactory)
+    {
+        var meter = meterFactory.Create("HatCo.Store");
+        _hatsSold = meter.CreateCounter<int>("hatco.store.hats_sold");
+        // Other methods in this class can now use the _hatsSold counter to increment the count.
+    }
+```
+
+Instead of adding the `IMeterFactory` to an existing type, you could also create a new type dedicated to managing some metrics. This pattern has a better separation of concerns
+in exchange for being a bit more verbose.
+
+```cs
+public class HatCoMetrics
+{
+    private readonly Counter<int> _hatsSold;
+
+    public HatCoMetrics(IMeterFactory meterFactory)
+    {
+        var meter = meterFactory.Create("HatCo.Store");
+        _hatsSold = meter.CreateCounter<int>("hatco.store.hats_sold");
+    }
+
+    public void HatsSold(int quantity)
+    {
+        _hatsSold.Add(quantity);
+    }
+}
+```
+
+Register the type with DI container in `Program.cs`.
+
+```cs
+var builder = WebApplication.CreateBuilder(args);
+builder.Services.AddSingleton<HatCoMetrics>();
+```
+
+Inject the metrics type and record values where needed. Because the metrics type is registered in DI it can be used with MVC controllers, minimal APIs, or any other type that is created by DI:
```

Comment:
```suggestion
Inject the metrics type and record values where needed. Because the metrics type is registered in DI, it can be used with MVC controllers, minimal APIs, or any other type that is created by DI:
```